src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMappingTest.java

```diff
@@ -385,22 +388,112 @@ void getHearingLocations() {
+            new Tuple[] {tuple(HEARING_VENUE_ID_1, COURT)});
+    }
+
+    @DisplayName("When a case has been adjourned and the same venue has been selected, return the same venue")
+    @Test
+    void getHearingLocationsAdjournmentSameVenue() {
+        SscsCaseData caseData = buildOverrideHearingLocations();
+
+        setupAppeal(caseData);
+        setupAdjournedHearingVenue(caseData, "sameVenue");
+
+        caseData.setHearings(Collections.singletonList(Hearing.builder()
+                    .value(uk.gov.hmcts.reform.sscs.ccd.domain.HearingDetails.builder()
+                    .venueId(HEARING_VENUE_ID_2).build())
+                .build()));
+
+        checkHearingLocationResults(HearingsDetailsMapping.getHearingLocations(caseData, referenceDataServiceHolder),
+            new Tuple[] {tuple(HEARING_VENUE_ID_2, COURT)});
+    }
+
+    @DisplayName("When a case has been adjourned but the next hearing is paper, return the override hearing locations")
+    @Test
+    void getHearingLocationsAdjournmentNewVenuePaperCase() {
+        SscsCaseData caseData = buildOverrideHearingLocations();
+
+        setupAdjournedHearingVenue(caseData, "somewhereElse");
+
+        checkHearingLocationResults(HearingsDetailsMapping.getHearingLocations(caseData, referenceDataServiceHolder),
+            new Tuple[] {tuple(HEARING_VENUE_ID_1, COURT), tuple(HEARING_VENUE_ID_2, COURT)});
+    }
+
+    @DisplayName("Checks both the errors we can throw when trying to obtain the venue ID when getting the locations")
+    @Test
+    void getHearingLocationsFailOnGettingVenueId() {
+        SscsCaseData caseData = buildOverrideHearingLocations();
+
+        setupAppeal(caseData);
+
+        String unknownValue = "UnknownValue";
+        setupAdjournedHearingVenue(caseData, "UnknownValue");
+
+        assertThatThrownBy(() -> HearingsDetailsMapping.getHearingLocations(caseData, referenceDataServiceHolder))
+            .isInstanceOf(IllegalStateException.class)
+            .hasMessageContaining("Failed to determine next hearing venue: " + unknownValue);
+
+        caseData.setAdjournCaseNextHearingVenue("sameVenue");
+
+        assertThatThrownBy(() -> HearingsDetailsMapping.getHearingLocations(caseData, referenceDataServiceHolder))
+            .isInstanceOf(IllegalStateException.class)
+            .hasMessageContaining("Failed to determine next hearing venue due to no latest hearing");
+    }
+
+    void setupAdjournedHearingVenue(SscsCaseData caseData, String nextHearingVenue) {
+        DynamicListItem item = new DynamicListItem(HEARING_VENUE_ID_1, "");
+        DynamicList list = new DynamicList(item, Collections.emptyList());
+
+        caseData.setAdjournCaseNextHearingVenue(nextHearingVenue);
+        caseData.setAdjournCaseNextHearingVenueSelected(list);
+    }
+
+    void checkHearingLocationResults(List<HearingLocation> hearingLocations, Tuple[] expectedResult) {
```

Comment:
This is good, but could be improved I think by not taking a Tuple as an arg (looks ugly being passed in, and one of the pair is always COURT). Try
```suggestion
    void checkHearingLocationResults(List<HearingLocation> hearingLocations, String... expectedIds) {
```
This allows the method to take any number of String args. Then you can build the tuple list within this method with a .map